garrysmod/lua/includes/modules/draw.lua

```diff
@@ -132,15 +134,15 @@ function MultiColorText( font, x, y, xAlign, yAlign, ... )
 	local w, h = 0, 0
 
 	if ( xAlign != TEXT_ALIGN_LEFT ) then
-		local fullText = ""
-		
+		table.Empty( MultiColorText_StringTable )
+
 		for k, v in ipairs, table, origin do
 			if not IsColor( v ) then
-				fullText = fullText .. tostring( v )
+				table.insert( MultiColorText_StringTable, tostring( v ) )
 			end
 		end
 
-		w, h = surface.GetTextSize( fullText )
+		w, h = surface.GetTextSize( table.concat( MultiColorText_StringTable, "" ) )

```

Comment:
Since you decided to upvalue the table this wont work anymore.
Imagine this, you had lots of strings before, but now less. This will merge previous strings with the new one, use `i` counter

Reply:
Ignore, see the first conversation